polygon-digital-carbon/src/utils/Token.ts

```diff
@@ -66,3 +70,33 @@ export function createICRTokenWithCall(tokenAddress: Address, tokenId: BigInt):
     token.save()
   }
 }
+
+export function loadOrCreateToken(tokenAddress: Address): Token {
+  let token = Token.load(tokenAddress)
+  log.debug('Loading token {}', [tokenAddress.toHexString()])
+  if (token == null) {
+    let tokenContract = ERC20.bind(tokenAddress)
+    token = new Token(tokenAddress)
+
+    let nameCall = tokenContract.try_name()
+    if (nameCall.reverted) token.name = ''
+    else token.name = nameCall.value
+
+    let symbolCall = tokenContract.try_symbol()
+    if (symbolCall.reverted) token.symbol = ''
+    else token.symbol = symbolCall.value
+
+    let decimalCall = tokenContract.try_decimals()
+    if (decimalCall.reverted) token.decimals = 18 // Default to 18 decimals
+    else token.decimals = decimalCall.value
+
+    token.latestPriceUSD = tokenAddress == USDC_ERC20_CONTRACT ? BigDecimal.fromString('1') : ZERO_BD
+    token.latestPriceUSDUpdated = ZERO_BI
+    token.latestPricePerKLIMA = ZERO_BD
+    token.latestPricePerKLIMAUpdated = ZERO_BI
+    token.isExAnte = false
```

Comment:
So right now no tokens will ever be labeled exAnte? We add that logic later?

Reply:
This function is only used for loadOrCreateCarbonPool, ICR has its own token creation function `createICRTokenWithCall` that will assign the correct exPost or exAnte